src/transformer.js

```diff
@@ -80,11 +80,15 @@ function transform(survey) {
           }
         : {};
 
-    const mediaMap = Object.fromEntries(
-        Object.entries(survey.media || {}).map((entry) =>
-            entry.map(escapeURLPath)
-        )
-    );
+    let mediaMap = null;
+
+    if (survey.media) {
+        mediaMap = Object.fromEntries(
+            Object.entries(survey.media).map((entry) =>
+                entry.map(escapeURLPath)
+            )
+        );
+    }
```

Comment:
Further on in the code some `if()` statements intended to bypass when mediaMap was null or falsy. This fix makes sure mediaMap is `null` if it is not provided.

Interestingly, Enketo Express never provides `survey.media` (so it's `undefined`). This means that this fix alone would fix the bug in Enketo Express.

I found some discussion about why mediaMap replacements are still in enketo-transformer though they are not used by us any more: https://github.com/enketo/enketo-transformer/pull/111

Reply:
Express is now doing its own media replacement but the `mediaMap` in Transformer was kept for backwards compatibility, is that right? Do we know for sure that it's used and/or should we consider putting it on a deprecation path? I remember this whole area was extremely confusing and any simplification that can be made would have value! It's surprising how many seemingly-unrelated issues have come out of changes in this area.

Reply:
Yes, I think it was [this comment](https://github.com/enketo/enketo-transformer/pull/111#issuecomment-881085153) by @eyelidlessness:

> It ensures the URLs produced are valid, which means they can also be used consistently in any environment.

It makes sense because it does make enketo-transformer better as a library. I don't have a strong opinion about removing it. The reason for placing enketo-transformer in a separate library was to facilitate development of hybrid mobile apps, and since then it turned out to also be useful for enketo-core development (and enketo-core-performance-monitor). I know that medic ended up not using it (but that may have changed and it likely has), and that ESRI (Survey123) was using it.

Reply:
Thank you for pulling the context out! Makes sense.